src/__tests__/pinning.test.js

```diff
@@ -1,398 +1,342 @@
+  }
+  pinning.openDB = myOpenDB
+  return pinning
 }
 
-jest.mock('redis', () => { return require('redis-mock') })
-
 describe('Pinning', () => {
+  let tmpDir
   let pinning
   let testClient
-  let analyticsMock
+  let clientIpfsOpts
 
-  jest.setTimeout(30000)
+  const pinningRoom = 'test-pinning-room'
 
-  beforeAll(async () => {
-    analyticsMock = {
-      trackPinDB: jest.fn(),
-      trackSyncDB: jest.fn(),
-      trackSpaceUpdate: jest.fn(),
-      trackPublicUpdate: jest.fn(),
-      trackRootUpdate: jest.fn(),
-      trackThreadUpdate: jest.fn(),
-      trackPrivateUpdate: jest.fn(),
-      trackPinDBAddress: jest.fn(),
-      trackSpaceUpdateByApp: jest.fn()
+  beforeEach(async () => {
+    tmpDir = await tmp.dir({ unsafeCleanup: true })
+    const orbitdbPath = tmpDir.path + '/orbitdb'
+    const ipfsPath = tmpDir.path + '/ipfs'
+    const ipfsOpts = {
+      config: pinningIpfsConfig,
+      repo: ipfsPath
     }
-    pinning = new Pinning({ repo: IPFS_PATH_1 }, ODB_PATH_1, analyticsMock, undefined, undefined, PINNING_ROOM)
-    testClient = new TestClient()
-    testClient.onMsg = jest.fn()
-
-    pinning._dbOpenedBefore = jest.fn().mockReturnValue(false)
-    await Promise.all([pinning.start(), testClient.init()])
-
-    register3idResolver()
+    const orbitCacheOpts = null
+    const pubSubConfig = null
+    const entriesNumCacheOpts = null
+    const pinWhitelistDids = null
+    const pinWhitelistSpaces = null
+    const pinSilent = null
+
+    pinning = new Pinning(ipfsOpts, orbitdbPath, analyticsMock, orbitCacheOpts, pubSubConfig, pinningRoom, entriesNumCacheOpts, pinWhitelistDids, pinWhitelistSpaces, pinSilent)
+    await pinning.start()
+    await pinning.entriesCache.store.flushall()
+    const pinningAddresses = await pinning.ipfs.swarm.localAddrs()
+    clientIpfsOpts = { config: { Bootstrap: pinningAddresses } }
+    testClient = new TestClient(clientIpfsOpts, pinningRoom)
+    await testClient.init()
+    await registerMock3idResolver()
+    pinning = addReplicatedEmitter(pinning)
   })
 
-  beforeEach(() => {
-    testClient.onMsg.mockClear()
-    cache.write.mockClear()
+  afterEach(async () => {
+    await testClient.cleanup()
+    await pinning.stop()
+    await tmpDir.cleanup()
   })
 
   it('should sync db correctly from client', async () => {
-    await testClient.createDB(true)
-    const responsesPromise = new Promise((resolve, reject) => {
-      const hasResponses = []
-      testClient.onMsg.mockImplementation((topic, data) => {
-        if (data.type === 'HAS_ENTRIES') {
-          expect(data.numEntries).toEqual(0)
-          if (hasResponses.indexOf(data.odbAddress) === -1) {
-            hasResponses.push(data.odbAddress)
-          }
+    await testClient.createDB(mockProfileData)
+    const pinningReplicatedPromise = new Promise((resolve) => {
+      const pinningStoreEntries = {}
+      const checkIfStoresReplicated = (data) => {
```

Comment:
I think checking if the data has actually replicated is great, but these tests where also testing that the pinning node actually sent the correct `HAS_ENTRIES` responses as well. Can we maybe check both?

Reply:
As I understood it from speaking with @zachferland , we no longer send an updated HAS_ENTRIES message on the first pin request, because it would create a lot of pubsub traffic. Do we want to test whether it sends an accurate one on a following pin request?

Reply:
It sends 0 on the first request? I think that's expected behavior now. At least, we need to confirm that it always sends a response.

Reply:
it sends it only as response to PIN_DB message, was saying we wouldnt want to send HAS_ENTRIES on receiving every new entry/update for any db (thats what that one line change did before), do you consider the PIN_DB message as the pin request?

Reply:
Right, that makes sense @zachferland

Reply:
To confirm, I should add to the test that a `HAS_ENTRIES` message is received by the client, ignoring the value for `numEntries`?

Reply:
Yeah, I think that should be enough for now.